Test/dafny0/TypeMembers.dfy

```diff
@@ -0,0 +1,199 @@
+// RUN: %dafny /compile:3 /dprint:"%t.dprint" "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+method Main() {
+  BasicTests();
+  MoreTests();
+}
+
+// ---------- basic tests ----------
+
+method BasicTests() {
+  var t := Yes;
+  var r := t.W();  // 10
+  var q := t.Q() + DaTy.Q();  // 13 + 13
```

Comment:
I'm surprised `t.Q()` works here.  I tried to find this in the reference manual but failed.  3.5.0 was the only relevant bit I could find.  Is it in there and I missed it, or should we update the manual?

Reply:
Dafny allows static members of a class to be referenced by an expression whose type is the class. Other than being used to look up the static member, the given receiver expression is completed ignored (so it may even be an expression that wouldn't be well-formed in other contexts).

It would be good to update the manual on this (and many other) point(s).